core/src/main/java/com/linecorp/armeria/common/AbstractContextAwareExecutor.java

```diff
@@ -73,7 +74,10 @@ public final E withoutContext() {
 
     final Runnable makeContextAware(Runnable task) {
         final RequestContext context = contextOrNull();
-        return context == null ? task : context.makeContextAware(task);
+        if (context != null) {
+            return context.makeContextAware(task);
+        }
+        return ArmeriaContextPropagation.captureAll().wrap(task);
```

Comment:
Question) Is `ArmeriaContextPropagation` invoked as a fallback only when `context == null`?

Reply:
I didn't want to capture the thread-local context twice since `ContextAwareRunnable already does this.

I understand the potential confusion, but unfortunately I don't think there is an easy way to clean this up unless `RequestContext` is also propagated with `ArmeriaContextPropagation`. <- this probably isn't possible without breaking changes though